### PR TITLE
List

### LINKED ISSUE #6: List Component
# User Story
As a **programmer**,
I want to **Implement a list component**,
so I can **Create a list which I want to use in further progress**.

# Description
image

#Material

# Done

- [x] Setup a basic input field
- [x] Display entered value below input field
- [x] Add hooks
- [x] Container for multiple lists
- [x] Create components
- [x] Create route for list item
- [x] Basic design

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
Fix #6 

> Added basic form to create and display lists

### CODE REVIEW

File: client/src/components/CreateListForm.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+function CreateListForm({ name, setName, cancelDisplay, handleSubmit }) {
```
Comment (marwinburesch) [inline client/src/components/CreateListForm.js:4]: An excellent idea! However, I would recommend renaming `cancelDispaly`. I don't think the name really describes what the function does, but more so what the effect for the user is. The function is used to ask the user for a name for the list, a side effect would be that the dialogue is shown.
I know this sounds like nit-picking, but in your `handleSubmit` later `cancelDisplay` will be called and then it's a little unclear what actually happens :)

File: client/src/components/CreateListForm.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        placeholder="Enter shopping list name"
+        onChange={(event) => setName(event.target.value)}
+      />
+
+      <button onClick={() => cancelDisplay()}>Cancel</button>
+      <input
+        type="submit"
+        //disabled={!name}
+        value="Create list"
+        palceholder="Create"
```
Comment (marwinburesch) [inline client/src/components/CreateListForm.js]: Does this really need a placeholder?

File: client/src/components/ListItemText.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+function ListItemText({ titel }) {
```
Comment (marwinburesch) [inline client/src/components/ListItemText.js]:
```suggestion
function ListItemText({ title }) {
```

File: client/src/components/ListItemText.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  return (
+    <div>
+      <p>1.1.2020</p>
+      <h4>{titel}</h4>
```
Comment (marwinburesch) [inline client/src/components/ListItemText.js]:
```suggestion
      <h4>{title}</h4>
```

**Review by marwinburesch — CHANGES_REQUESTED** [review]
Excellent work! I think this PR has a good size, great idea to push the styling to a different PR. Just a few minor things :)

- I would rename `client/src/api/data.js` to `lists.js`

- Please choose a convention regarding the position of your styled components, in terms of before or after the main component

Cheers, carry on 🐜

File: client/src/components/CreateListForm.js (hunk 1 above)
Comment (LebkuchenCJ) [reply client/src/components/CreateListForm.js:4]: Thanks for your comments. I will change them right away. 👍

**Pull request merged**